com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceInProgressOperationsExecution.java

```diff
@@ -0,0 +1,84 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+import org.flowable.engine.delegate.DelegateExecution;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationState;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceOperationGetter;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceProgressReporter;
+import com.sap.cloud.lm.sl.cf.process.util.StepLogger;
+import com.sap.cloud.lm.sl.common.SLException;
+
+public class PollServiceInProgressOperationsExecution extends PollServiceOperationsExecution {
+
+    public PollServiceInProgressOperationsExecution(ServiceOperationGetter serviceOperationGetter,
+                                                    ServiceProgressReporter serviceProgressReporter) {
+        super(serviceOperationGetter, serviceProgressReporter);
+    }
+
+    @Override
+    protected List<CloudServiceExtended> getServicesData(DelegateExecution context) {
+        return StepsUtil.getServicesData(context);
+    }
+
+    @Override
+    protected void reportServiceState(ExecutionWrapper execution, CloudServiceExtended service, ServiceOperation lastServiceOperation) {
```

Comment:
I think the reporting is the same on the three places?
Could we try to unify it?